packages/cli/src/services/annotationTag.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Service } from 'typedi';
+import { validateEntity } from '@/GenericHelpers';
+import type { ITagWithCountDb } from '@/Interfaces';
+import type { AnnotationTagEntity } from '@db/entities/AnnotationTagEntity';
+import { AnnotationTagRepository } from '@db/repositories/annotationTag.repository';
+import { ExternalHooks } from '@/ExternalHooks';
+
+type GetAllResult<T> = T extends { withUsageCount: true }
+	? ITagWithCountDb[]
+	: AnnotationTagEntity[];
```

Comment:
I think this type is wrong, as we only return id, name, createdAt and updatedAt, but not annotations or annotationMappings

Reply:
Refactored this a bit, please take a look